public/javascript/_common/services/providers/thirdPartyProviders/soundcloud/search/soundcloudSearch.service.js

```diff
@@ -0,0 +1,88 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('ammo.soundcloud.search.service', [
+      'ammo.constants.service'
+    ])
+    .factory('soundcloudSearch', soundcloudSearchService);
+
+  function soundcloudSearchService($http, $q, constants) {
+    var service;
+
+    service = {
+      search: search
+    };
+
+    return service;
+
+    ///////////
+    /**
+     * @param {string} query Query user typed on the search box
+     * @param {number} limit Limit the number of search results (integer)
+     * @returns {promise} Promise that will resolve to an array of song objects
+     */
+    function search(query, limit) {
+      var deferred = $q.defer();
```

Comment:
do we still need deferred here? I think the reason we needed it for youtube was because of the 2 requests thing